tensorflow_model_optimization/g3doc/guide/clustering/clustering_with_keras.ipynb

```diff
@@ -0,0 +1,1164 @@
+    "2.   Train a clustered model with the MNIST dataset with the clustering API.\n",
+    "3.   Compare the size of the clustered model and the non-clustered one after compression.\n",
+    "4.   Convert the clustered model to TensorFlow Lite format and verify the accuracy.\n",
+    "5.   Show how the clustered model works with other optimization techniques, for instance, post-training quantization.\n",
+    "\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "To use the clustering API, install the `tensorflow-model-optimization` package. See the [TensorFlow model optimization repo](https://github.com/tensorflow/model-optimization) for compatible API versions.\n",
+    "\n",
+    "Since you will train a few models in the tutorial, install the `tensorflow-gpu` package to speed things up. Enable the GPU with: *Runtime > Change runtime type > Hardware accelator* and make sure GPU is selected."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "! pip uninstall -y tensorflow\n",
+    "! pip uninstall -y tf-nightly-gpu\n",
+    "#! pip install tensorflow-gpu==2.0.0-beta\n",
+    "! pip install -U tensorflow-gpu==1.15.0\n",
+    "\n",
+    "#Please note: during the review time of the PR, make sure you install the master branch of \n",
+    "#tensorflow=model-optimization to use our clustering API from\n",
+    "#here https://github.com/tensorflow/model-optimization#installing-from-source\n",
+    "#use this once the PR is approved\n",
+    "#! pip install tensorflow-model-optimization"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext tensorboard\n",
+    "import tensorboard\n",
+    "\n",
+    "import tensorflow as tf\n",
+    "\n",
+    "import tempfile\n",
+    "import zipfile\n",
+    "import os\n",
+    "import sys\n",
+    "\n",
+    "from tensorflow.python import keras"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prepare the training data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "x_train shape: (60000, 28, 28, 1)\n",
+      "60000 train samples\n",
+      "10000 test samples\n"
+     ]
+    }
+   ],
+   "source": [
+    "batch_size = 128\n",
+    "num_classes = 10\n",
+    "epochs = 12\n",
+    "\n",
+    "# input image dimensions\n",
+    "img_rows, img_cols = 28, 28\n",
+    "\n",
+    "# the data, shuffled and split between train and test sets\n",
+    "(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()\n",
+    "\n",
+    "if tf.keras.backend.image_data_format() == 'channels_first':\n",
+    "  x_train = x_train.reshape(x_train.shape[0], 1, img_rows, img_cols)\n",
```

Comment:
Should we run a checker (e.g. linter) for the jupyter notebook?

Reply:
From the TensorFlow Docs team: https://github.com/tensorflow/docs/blob/master/tools/nbfmt.py is a WIP linter. Seeing if there are plans to also lint the python code in accordance to https://www.tensorflow.org/community/contribute/code_style.